### PR TITLE
feat: Add method to retrieve all listeners from node

### PR DESCRIPTION
**Pull request by Distractic**
Implementation for #1592

### CODE REVIEW

File: src/main/java/net/minestom/server/event/EventNode.java [hunk 1]
```diff
@@ -332,6 +333,13 @@ default void removeChildren(@NotNull String name) {
     @Contract(value = "_ -> this")
     @NotNull EventNode<T> removeListener(@NotNull EventListener<? extends T> listener);
 
+    /**
+     * Get all listeners registered to this node.
+     * @return A collection of listeners.
+     */
+    @Contract(value = "-> this")
+    @NotNull Collection<EventListener<T>> getListeners();
```
Comment (TheMode) [inline src/main/java/net/minestom/server/event/EventNode.java:341]: The `Contract` is wrong. Also `listeners()` would be a better name

**Review by TheMode — CHANGES_REQUESTED** [review]
I am not sure to be convinced, especially given the slowness of the getter

File: src/main/java/net/minestom/server/event/EventNode.java (hunk 1 above)
Comment (Distractic) [reply src/main/java/net/minestom/server/event/EventNode.java]: I removed the contract and change the method's name.

**Comment by Distractic** [comment]
> I am not sure to be convinced, especially given the slowness of the getter

Sure, this method should be called exceptionally and it's my use case
After, if you have an idea to get listeners with a faster implementation, I'm interested

**Comment by NitrinCloud** [comment]
> > I am not sure to be convinced, especially given the slowness of the getter
> 
> Sure, this method should be called exceptionally and it's my use case After, if you have an idea to get listeners with a faster implementation, I'm interested

You could use 2 for loops instead of streams

**Comment by mworzala** [comment]
Closing with the issue, can be reopened later if more discussion happens

**Pull request closed without merging**